projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3483,6 +3487,7 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
         this._pinnedRowList.changes
             .pipe(takeUntil(this.destroy$))
             .subscribe((change: QueryList<IgxGridRowComponent>) => {
+                this.buildDataView();
```

Comment:
this  might need to be done in a different handler or in the pipes. I remember you saying that there was an issue with sorting with pinned rows and I believe it could be because rows are reused which causes the QueryList to not fire `changes` when its simply reordered.

I remember this is why we e.g. order the virtual rows on the rowList getter  by index, because the list itself doesn't change when we reorder the views around during scrolling.